updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,222 @@
+The MCP for the control plane is called master pool and other group which have worker nodes is called worker pool.
+During an OpenShift update all pools are updated concurrently.
+
+Nodes within a MCP are updated by cordoning and draining up to the specified `maxUnavailable` number of nodes (if required).
+Once drained the Machine Config Daemon applies MachineConfig changes which may include updating the OS and the host is rebooted.
+The MCP gives us the ability to pause a pool and it will make sure that the OS is not updated/rebooted during the update process.
+We can create more than one customized machine config pool out of the worker pool, which will give us a more control over
+the sequence in which the nodes are updated.
+
+[NOTE]
+====
+Creating custom MCP using master nodes is not supported.
+Machine config operator (responsible for updating the nodes) will ignore the custom MCP created out of the master nodes.
+This is required to make sure control plane nodes remain stable.
+====
+
+An OpenShift or Kubernetes cluster is highly available by design and there are many Kubernetes features
+(e.g. pod disruption budget, pod affinity, health checks, replicas) which can make the application highly available in case of node failures.
+
+However there might be some secenarios where a controlled rollout of upgrade to the worker nodes is desired.
+This helps to make sure the mission critical application stays available during the whole update process even if the upgrade process causes a failure.
+It is also useful in scenarios where we can not have a single maintenance window to complete the whole update process.
+
+The slow rollout of a new OpenShift version to worker nodes can be characterized as a canary release i.e. we can
+control the rollout of worker nodes by controlling the machine config pools. After the update of the first MCP, the
+application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+
+== Workflow for updating worker nodes with canary rollout
+
+. Create MCPs out of the worker pool. The number of nodes in each MCP depends on a few factors like maintaince window duration for each MCP and the amount of reserve capacity (extra worker nodes) present.
++
+[NOTE]
+====
+In case of a failure i.e. if the MCP with the new version does not work as expected with the applications, the nodes in the pool can be cordoned and drained. So that the application will not run on these nodes and the extra capacity will help to maintain the quality of service of the applications.
+====
++
+. Pause the MCPs you do not want to update as part of the default update process.
++
+[NOTE]
+====
+Pausing the MCP will also pause the kube-apiserver-to-kubelet-signer automatic CA certificates rotation.
+New CA certificates are generation and removal happens at 292 day (from the installation day) and 365 days respectively.
+The renewal and removal of the certs need reboot of the nodes.
```

Comment:
With OCP 4.7 and onward, cert rotation doesn't require node reboot or drain https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigDaemon.md#drainless-and-rebootless-updates . Unpausing MCP is required so that new rendered config that contains updated cert content can be applied on node without drain and reboot. If this doc is intended to be available for 4.7 or above releases, we can update it with same. Otherwise, we should mention behavior for both cases OCP_REL < 4.7 and OCP_REL >= 4.7.

Reply:
Updated the text to reflect your suggestion. I wrote the doc in a way that we can back port it to till 4.6 without any changes. Also I am little skeptical about writing too much about the cert rotation here. Ideally there should be some documentation about the automatic cert rotation in the OCP official docs which I could have linked here.

Reply:
Pushed the change in https://github.com/openshift/openshift-docs/commit/8e7ab60fab996fa62e4fae1dbff4dfa0cfd75e27

Reply:
agree that this is very confusing and everything is not documented at single place. We all have realized this behavior few months back when cluster started completing 1 year lifecycle and at the same time pools are paused. I believe paused pool was never implemented as a feature and it came as a escape hatch.